bip-xxxx.mediawiki

```diff
@@ -0,0 +1,358 @@
+
+===Optional parameters===
+
+When the payjoin sender posts the original PSBT to the receiver, he can optionally specify the following HTTP query string parameters:
+
+* <code>v=</code>, the version number of the payjoin protocol that the sender is using. The current version is <code>1</code>.
+
+This can be used in the future so the receiver can reject a payjoin if the sender is using a version which is not supported via an error HTTP 400, <code>version-unsupported</code>.
+If not specified, the receiver will assume the sender is <code>v=1</code>.
+
+If the receiver does not support the version of the sender, they should send an error with the list of supported versions:
+<pre>
+{
+    "errorCode": "version-unsupported",
+    "supported" : [ 2, 3, 4 ],
+    "message": "The version is not supported anymore"
+}
+</pre>
+
+* <code>additionalfeeoutputindex=</code>, the preferred output from which to increase the fee for the added inputs. (default: <code>-1</code>)
+
+If the <code>additionalfeeoutputindex</code> is out of bounds or pointing to the payment output meant for the receiver, the receiver should ignore the parameter.
+Should be ignored in the [[#spare-change|spare change]] case.
+
+* <code>maxadditionalfeecontribution=</code>, an integer defining the maximum amount in satoshis that the sender is willing to contribute towards fees for the additional inputs. <code>maxadditionalfeecontribution</code> must be ignored if set to less than zero. (default: -1)
+
+Note that if <code>maxadditionalfeecontribution</code> is too low, the sender should create a transaction with RBF disabled, as the original transaction could replace the payjoin transaction.
+Should be ignored in the [[#spare-change|spare change]] case.
+
+* <code>minfeerate=</code>, a decimal in satoshi per vbyte that the sender can use to constraint the receiver to not drop the minimum fee rate too much.
+
+Should be ignored in the [[#spare-change|spare change]] case.
+
+==Rationale==
+
+There is several consequences of our proposal:
+
+* The receiver can bump the fee of the original transaction.
+* The receiver can modify the outputs of the original PSBT.
+* The sender must provide the UTXO information (Witness or previous transaction) in the PSBT.
+
+===Respecting the minimum relay fee policy===
+
+To be properly relayed, a Bitcoin transaction needs to pay at least 1 satoshi per virtual byte.
+When fees are low, the original transaction is already 1 satoshi per virtual byte, so if the receiver adds their own input, they need to make sure the fee is increased such that the rate does not drop below 1 satoshi per virtual byte.
+
+===Preventing mempool replacement===
+
+A safe way to implement payjoin, is for both the sender and receiver to try broadcasting the original transaction at some fixed interval period regardless of the state of the payjoin.
```

Comment:
By broadcasting the original transaction, you are revealing information about the ownership of the inputs and outputs. So I would say that broadcasting the original transaction is something you should do as a last resort if the sender fails to sign and publish the PayJoin transaction or if the PayJoin transaction is failing to get mined by the network and the original transaction has a higher fee rate than the PayJoin transaction. I liked the way this was [documented](https://github.com/bitcoin/bips/blob/master/bip-0079.mediawiki#transaction-publishing) in BustaPay. The same comment applies to the section "Receiver does not need to be a full node", where it is stated that a receiver can "automatically broadcast the original transaction after a timeout of 1 minute".

Of course if the PayJoin transaction is good, then the original transaction is revealed only to a limited set of individuals and won't go on permanent record, so it's not a deal breaker, but I don't think it should be encouraged "regardless of the state of the payjoin". The participants should determine the state of the PayJoin and then decide whether to broadcast.

Reply:
>  The participants should determine the state of the PayJoin and then decide whether to broadcast.

This is very tricky code here. What about I change it to: "A safe way to implement payjoin if you have a full node"

Reply:
I guess what I don't understand is the concern about being a full node or not. To me this topic seems to be out of scope of this document. In order for the recipient to decide whether to for example provide the service that was the subject of the payment, the recipient must have some means to determine whether he "received" the payment, i.e. whether the transaction in question was mined and perhaps its current depth. How he does that or whom he trusts is up to him. So in order to be able to operate, the recipient must have a means to determine the state of a transaction, which is why I fail to see the need for automatic broadcasts.